src/app/delegates/components/IssuesFilter.tsx

```diff
@@ -0,0 +1,90 @@
+"use client";
+
+import { useRouter, useSearchParams } from "next/navigation";
+import { Listbox } from "@headlessui/react";
+import { Fragment } from "react";
+import { ChevronDown } from "lucide-react";
+import { useAddSearchParam, useDeleteSearchParam } from "@/hooks";
+import { useAgoraContext } from "@/contexts/AgoraContext";
+import Tenant from "@/lib/tenant/tenant";
+
+const FILTER_PARAM = "issueFilter";
+const DEFAULT_FILTER = "all";
+
+export default function IssuesFilter() {
+  const { ui } = Tenant.current();
+
+  const router = useRouter();
+  const searchParams = useSearchParams();
+  const addSearchParam = useAddSearchParam();
+  const deleteSearchParam = useDeleteSearchParam();
+  const filterParam = searchParams?.get(FILTER_PARAM) || "all";
+  const { setIsDelegatesFiltering } = useAgoraContext();
```

Comment:
I feel like the agoraContext thing is an anti-pattern, feels weird to me that we are stuffing various pieces of state in this context provider